File: benchmark/adler32/adler32.lua
local function readAll(file)
    local f = assert(io.open(file, "rb"))
    local content = f:read("*all")
    f:close()
    return content
end

local function adler32(bv)
      local a = 1
      local b = 0
      for i = 1,#bv do
          a = (a + bv:byte(i,i)) % 65521
          b = (a + b) % 65521
      end
      return a + (b * 65536)
end

print(string.format("%8.8X",adler32(readAll("test-files/r5rs.pdf"))))

File: benchmark/adler32/adler32.py
def adler32(v):
    a = 1
    b = 0
    for c in v:
        a = (a + c) % 65521
        b = (a + b) % 65521
    return a | (b << 16)

def readFile(filename):
    in_file = open(filename, "rb")
    data = in_file.read()
    in_file.close()
    return data

def adler32sum(filename):
    sum = adler32(readFile(filename))
    print("{0:08X}".format(sum))

adler32sum("test-files/r5rs.pdf")
